### PR TITLE
statev2: proto: Add proto definitions for all existing state transitions

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR adds protobuf definitions for all state types that currently have setter methods in the existing state implementation.

As well, this PR derives a builder pattern for all generated protobuf types.

### Testing
- Unit and integration tests pass
- Tested the builder pattern

### CODE REVIEW

File: statev2/proto/src/lib.rs [hunk 1]
```diff
@@ -1,28 +1,94 @@
...
 #[cfg(test)]
 mod tests {
-    use super::NewPeer;
+    use crate::{AddOrderBuilder, AddPeersBuilder, NetworkOrderBuilder, PeerInfoBuilder};
+
+    use super::{AddOrder, AddPeers};
     use prost::Message;
 
     /// Tests the add new peer message
     #[test]
```
Comment (akirillo) [inline statev2/proto/src/lib.rs:64]: is the plan to add serde tests for all proto definitions? feels weird to have it just be for some types, but can also just leave the serde code path to be covered by future unit tests of the state module

**Review by akirillo — APPROVED** [review]
lgtm, 2 small qs

File: statev2/proto/proto/orders.proto [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    ProtoUuid id = 1;
+    /// The public shares nullifier of the wallet holding this order
+    ProtoScalar nullifier = 2;
+    /// The cluster managing the order
+    ClusterId cluster = 3;
+    /// The state of the order in the network
+    NetworkOrderState state = 4;
+    /// The proof payload of `VALID COMMITMENTS` for this order
+    /// and `VALID REBLIND` for the containing wallet
+    bytes proof = 5;
```
[the remark this answers is not in the document]
Comment (joeykraut) [reply statev2/proto/proto/orders.proto:41]: not as far as i see :)

File: statev2/proto/src/lib.rs (hunk 1 above)
Comment (joeykraut) [reply statev2/proto/src/lib.rs:64]: No, at that point we're just testing the library, these are more sanity checks that the `prost` configuration ends up with what we want. No need to over-test

**Pull request merged**